https://issues.redhat.com/browse/ACM-6946 Updates after PM review

dockerymick opened the pull request:
https://issues.redhat.com/browse/ACM-6946
https://issues.redhat.com/browse/ACM-7540

dockerymick committed:
Updates after PM review

berenss commented on observability/observe_environments_intro.adoc:
```diff
@@ -2,7 +2,7 @@
 //Need to also complete modular effort | MJ | 05/02/23
 
-The observability component is a service you can use to understand the state of your clusters. The observability service is deployed by the `multiclusterhub-operator` pod, which is complete during the {product-title-short} installation. By default, observability is enabled on your hub cluster.
+The observability component is a service you can use to understand the state of your clusters. The observability service is deployed by the `multiclusterhub-operator` pod, which is complete during the {product-title-short} installation. By default, observability is disabled on your hub cluster.
```
The observability component is a service you can use to understand the health and utilization of clusters across your fleet. By default, the Multicluster Observability operator is enabled during the installation of {product-title-short}. As the  observability component requires a valid object store for metrics storage, it is necessary to follow the steps to configure observability on your hub cluster before you can begin using it. See "1.2. Enabling observability service"

berenss replied:
..to understand the health and utilization of clusters across your fleet.

berenss replied:
Multicluster Observability operator

dockerymick committed:
Alot of updates

dockerymick committed:
Updates

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -1,10 +1,14 @@
-
+//which of these topics should be moved to Using observability?
+//Customize: modifications to the service
+//Configure: arrange or order the service task
```
```suggestion
//Configure: arrange or order the service task

```

dockerymick committed:
Update observability/customize_observability.adoc

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -477,141 +466,48 @@ The observability service is enabled. After you enable the observability service
 
 [#enabling-observability-ocp]
-== Enabling observability from the {ocp} console
+=== Enabling observability from the {ocp} console
```
This will address the first request from ACM-7540

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,129 @@
+
+oc -n open-cluster-management-observability create secret tls proxy-byo-cert --cert ./ingress.crt --key ./ingress.key
+----
+
```
```suggestion

//this seems more like a configuration section
```

dockerymick committed:
Update observe_environments.adoc

moving content

dockerymick committed:
Update use_observability.adoc

updates

dockerymick committed:
Update customize_observability.adoc

updates

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -1,26 +1,19 @@
 
 Collect logs about new information that is created for observability resources with the `must-gather` command. For more information, see the _Must-gather_ section in the _Troubleshooting documentation_.
 
+//which of these topics should be moved to Using observability?
```
```suggestion
//I moved some topics to Using observability, need to confirm if changes make sense
```

dockerymick committed:
Update whats_new.adoc

moving observability out of the Web console section

dockerymick commented on release_notes/whats_new.adoc:
```diff
@@ -50,6 +46,17 @@ View release notes, as well as tasks and support information at link:../clusters
 For other Application topics, see link:../applications/app_management_overview.adoc#managing-applications[Managing applications].
 
+[#observability-whats-new]
+== Observability
```
This was added to address Scott's comment on issue [ACM-6946](https://issues.redhat.com/browse/ACM-6946).

"Observability is not a Console feature. It is not part of the console itself, as it is a separate link that launches a new window which runs Grafana for visualizations. 
Generally speaking, we are likely to have updates/release notes into Observability more often than in the Console proper.

So, the Multicluster Observability Operator is enabled by default - this is a NO-OP to the end user; this means this is working like it did before, just a repackaging essentially.
What remains true is that the customer needs to Enable Multicluster Observability by following the usual doc topic : https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/observability/observing-environments-intro#enabling-observability-service

we want to make sure the user does not get confused by the new wording - the wording makes it seem like they would not need to run the steps in 1.2.2"

dockerymick commented on observability/observe_environments_intro.adoc:
```diff
@@ -1,8 +1,9 @@
+//I noticed that multicluster observability operator is capital in the comments, is that how the operator name is displayed from the console?
 
-The observability component is a service you can use to understand the state of your clusters. The observability service is deployed by the `multiclusterhub-operator` pod, which is complete during the {product-title-short} installation. By default, observability is enabled on your hub cluster.
+The observability component is a service you can use to understand the health and utilization of clusters across your fleet. By default, multicluster observability operator is enabled during the installation of {product-title-short}. 
```
This update also addresses the comment from Scott in [ACM-6946](https://issues.redhat.com/browse/ACM-6946):

"So, the Multicluster Observability Operator is enabled by default - this is a NO-OP to the end user; this means this is working like it did before, just a repackaging essentially.
What remains true is that the customer needs to Enable Multicluster Observability by following the usual doc topic : https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html/observability/observing-environments-intro#enabling-observability-service

we want to make sure the user does not get confused by the new wording - the wording makes it seem like they would not need to run the steps in 1.2.2"

dockerymick replied:
Resolved during live review

dockerymick committed:
Update observability_enable.adoc

moved the disabling content

dockerymick committed:
Update customize_observability.adoc

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -210,6 +210,9 @@ Generating access keys using AWS Security Service require the following addition
 * Specify annotations for the observability service accounts that requires access to the S3 bucket. You can find an example of how observability on Red Hat OpenShift Service on AWS (ROSA) cluster can be configured to work with AWS STS tokens in the _Set environment_ step. See link:https://www.rosaworkshop.io/[Red Hat OpenShift Service on AWS (ROSA)] for more details, along with link:https://www.r ...
 
+[generate-access-keys]
+=== Generating access keys using the AWS Security Service
```
This change addresses request 3. The info for GCP is on line 107. I will verify that it is updated in 2.8 and earlier

Verified that Google Cloud Platform is present in our 2.8 docs: https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.8/html-single/observability/index

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -485,133 +476,68 @@ Create your object storage secret named, `thanos-object-storage` in the `open-cl
-----
-oc get route rbac-query-proxy -n open-cluster-management-observability
-----
+[#verifying-thanos-version]
```
Verify if this is a good location for this. I need help to determine where this step should be placed. There are a lot of substeps. After i receive dev confirmation about the steps, it will be more clear where to place the step

dockerymick commented on observability/observability_enable.adoc:
```diff
@@ -485,133 +476,68 @@ Create your object storage secret named, `thanos-object-storage` in the `open-cl
 
-The collection rules are grouped together as a parameter section named `collect_rules`, where it can be enabled or disabled as a group. {product-title-short} installation includes the collection rule group, `SNOResourceUsage` with two default collection rules: `HighCPUUsage` and `HighMemoryUsage`. The `HighCPUUsage` collection rule begins when the node CPU usage exceeds 70%. The `HighMemoryUsage` ...
+[#removing-observability-resource]
+== Removing observability
```
This change is related to the 4th request from 7540

dockerymick committed:
Update observability_arch.adoc

dockerymick committed:
Update observe_environments_intro.adoc

updating the order

dockerymick committed:
Update observe_environments.adoc

updating the title reference

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,178 @@
+[#additional-resources-use]
+== Additional resources
+
+- For more information, see link:https://prometheus.io/docs/concepts/remote_write_spec/[Prometheus Remote-Write specification].
```
```suggestion
- For more information, see link:https://prometheus.io/docs/concepts/remote_write_spec/[Prometheus Remote-Write specification].

- For more topics, return to xref:../observability/observe_environments_intro.adoc#observing-environments-intro[Observability service introduction].
```

dockerymick committed:
Update observability/use_observability.adoc

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,180 @@
++
+*Note:* The `QUERY_EXPRESSION` is the standard Prometheus query expression. For example, query the metrics `cluster_infrastructure_provider` by replacing the URL in the previously mentioned command with the following URL: `https://{PROXY_ROUTE_URL}/api/v1/query?query=cluster_infrastructure_provider`. For more details, see link:https://prometheus.io/docs/prometheus/latest/querying/basics/[Querying ...
+//IS THIS ACCURATE? THIS FILE WAS REMOVED
+//* You can also replace certificates for the `rbac-query-proxy` route. See link:../governance/cert_mgmt_ingress.adoc#openssl-commands-for-generating-a-certificate[OpenSSL commands for generating a certificate] to create certificates. When you customize the `csr.cnf`, update the `DNS.1` to the hostname for the `rbac-query-proxy` route.
```
Need to address this hidden comment.

dockerymick replied:
This was addressed during the live review

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,180 @@
++
+The `ep.yaml` is the key of the content and is used in the `MultiClusterObservability` custom resource in next step. Currently, observability supports exporting metrics to endpoints without any security checks, with basic authentication or with `tls` enablement. View the following tables for a full list of supported parameters:
+
+//this information is similar syntax used in the API docs. This seems out of place 
```
this needs to be addressed

dockerymick replied:
addressed during live review

dockerymick committed:
Update observability/customize_observability.adoc

dockerymick committed:
Update observability/use_observability.adoc

dockerymick committed:
updates

dockerymick committed:
Update customize_observability.adoc

dockerymick committed:
Update observe_environments.adoc

dockerymick committed:
Update use_observability.adoc

dockerymick committed:
Update customize_observability.adoc

dockerymick committed:
updates

dockerymick committed:
updates

dockerymick committed:
more updates

dockerymick commented on clusters/cluster_lifecycle/single_node_ocp.adoc:
```diff
@@ -49,16 +49,16 @@ skipMachinePools: true
 [#dynamic-metrics-for-sno]
 == Dynamic metrics for SNO clusters
-
+//this information is also in the customize_observability.adoc file should this be removed for the Customize obseravbility file? 
```
@berenss @bjoydeep @subbarao-meduri It seems like it would be best to add a link reference to this file specifically (single_node_ocp.adoc) from the Customizing observability configuration doc (customize_observability.adoc)

dockerymick replied:
```suggestion
//this information is also in the customize_observability.adoc file should this be removed from the Customize obseravbility file? 
```

dockerymick committed:
Update clusters/cluster_lifecycle/single_node_ocp.adoc

dockerymick commented on clusters/cluster_lifecycle/single_node_ocp.adoc:
```diff
@@ -49,16 +49,16 @@ skipMachinePools: true
 [#dynamic-metrics-for-sno]
 == Dynamic metrics for SNO clusters
-
+//this information is also in the customize_observability.adoc file should this be removed from the Customize obseravbility file? 
```
@berenss @bjoydeep @subbarao-meduri It seems like it would be best to add a link reference to this file specifically (single_node_ocp.adoc) from the Customizing observability configuration doc (customize_observability.adoc)

swopebe replied:
I think we need to keep the Create a Single-Node Openshift topic, though that is a task and it has Dynamic metrics in the middle of it.

It seems creating is for all of cluster, Dynamic is just for observability? If that is the case, we have an observability topic inside MCE, which could present problems since that is more for RHACM.

dockerymick commented on observability/observe_environments_intro.adoc:
```diff
@@ -1,8 +1,9 @@
 [#observing-environments-intro]
 = Observability service introduction
-//Need to also complete modular effort | MJ | 05/02/23
+//Starting on modular changes and improvements
+//I noticed that multicluster observability operator is capital in the comments, is that how the operator name is displayed from the console?
```
```suggestion

```

dockerymick committed:
Update observability/observe_environments_intro.adoc

dockerymick committed:
more improvements

swopebe committed:
Update single_node_ocp.adoc

swopebe commented on observability/observe_environments_intro.adoc:
```diff
@@ -19,6 +18,10 @@ Read the following documentation for more details about the observability compon
+* xref:../observability/use_observability.adoc#using-observability[Using observability]
+* xref:../observability/customize_observability#customizing-observability[Customizing observability]
+* xref:../observability/observability_alerts.adoc#observability-alerts[Observability alerts]
 * xref:../observability/search_intro.adoc#searching-in-the-console-intro[Searching in the console introduction]
```
I see there is enable, use, customize....

I think the word "use" can maybe be replaced with something more specific.

swopebe commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,194 @@
+oc get route rbac-query-proxy -n open-cluster-management-observability
+----
+
+* To access the `rbac-query-proxy` route, you must have an OpenShift OAuth access token. The token should be associated with a user or service account, which has permission to get namespaces. For more information, see link:https://docs.openshift.com/container-platform/4.11/authentication/managing-oauth-access-tokens.html[Managing user-owned OAuth access tokens].
```
Should these be steps and not bullets?

swopebe commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,194 @@
+*Note:* The `QUERY_EXPRESSION` is the standard Prometheus query expression. For example, query the metrics `cluster_infrastructure_provider` by replacing the URL in the previously mentioned command with the following URL: `https://{PROXY_ROUTE_URL}/api/v1/query?query=cluster_infrastructure_provider`. For more details, see link:https://prometheus.io/docs/prometheus/latest/querying/basics/[Querying ...
+
+** Run the following command to create `proxy-byo-ca` and `proxy-byo-cert` secrets using the generated certificates:
++
```
I think these should be broken into two commands/steps.

dockerymick replied:
done

swopebe commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,194 @@
+- <<viewing-and-exploring-data,Viewing and exploring data>>
+
+[#external-metric-query]
+== Using the external metric query
```
Accessing?

swopebe commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,194 @@
+[#exporting-metrics-to-external-endpoints]
+== Exporting metrics to external endpoints
+
+Use observability to export metrics to external endpoints, which support the Prometheus Remote-Write specification in real time. Complete the following steps to export metrics to external endpoints: 
```
Use observability to export metrics to external endpoints

change to

Export metrics to external endpoints

swopebe commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,194 @@
+
+*Note:* Do not set the `ObservabilitySpec.resources.CPU.limits` parameter if observability is enabled on single node managed clusters. When you set the CPU limits, it causes the observability pod to be counted against the capacity for your managed cluster. See the reference for _Management Workload Partitioning_ in the _Additional resources_ section.
+
+[#acm-provided-dashboards]
```
I think this may need to be a separate topic.

swopebe replied:
Viewing Dashboards seems to contain enough info to stand on it's own?

dockerymick replied:
discussed during live review

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -132,132 +147,47 @@ metadata:
-- The `uwl_metrics_list.yaml` is the key for the ConfigMap data.
-
-- The value of the ConfigMap data is in YAML format. The `names` section includes the list of metric names, which you want to collect from the `test` namespace. After you create the ConfigMap, the specified metrics from the target namespace is collected by the observability collector and pushed to the hub cluster.
+<1> The `uwl_metrics_list.yaml` is the key for the ConfigMap data.
```
```suggestion
<1> Enter the key for the ConfigMap data.
```

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -132,132 +147,47 @@ metadata:
-
-- The value of the ConfigMap data is in YAML format. The `names` section includes the list of metric names, which you want to collect from the `test` namespace. After you create the ConfigMap, the specified metrics from the target namespace is collected by the observability collector and pushed to the hub cluster.
+<1> The `uwl_metrics_list.yaml` is the key for the ConfigMap data.
+<2> The value of the ConfigMap data is in YAML format. The `names` section includes the list of metric names, which you want to collect from the `test` namespace. After you create the ConfigMap, the specified metrics from the target namespace is collected by the observability collector and pushed to the hub cluster.
```
```suggestion
<2> Enter the value of the ConfigMap data in YAML format. The `names` section includes the list of metric names, which you want to collect from the `test` namespace. After you create the ConfigMap, the specified metrics from the target namespace is collected by the observability collector and pushed to the hub cluster.
```

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -277,10 +207,93 @@ spec:
+
+[#dynamic-metrics-for-sno]
+=== Dynamic metrics for single-node OpenShift clusters
+//recommend to remove this info and add reference link to the same section that is within the cluster lifecycle folder. | MJ | 09/29/23
```
@subbarao-meduri can you remind me if Is this applicable for both MCE and ACM?

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -277,10 +207,93 @@ spec:
 
 [#updating-replicas]
-== Updating the _MultiClusterObservability_ custom resource replicas from the console
+== Updating the MultiClusterObservability custom resource replicas from the console
```
```suggestion
== Updating the _MultiClusterObservability_ custom resource replicas from the console
```

dockerymick commented on observability/design_grafana.adoc:
```diff
@@ -168,6 +156,8 @@ clusterrolebinding.rbac.authorization.k8s.io "open-cluster-management:grafana-cr
 
 [#additional-resource-grafana]
 == Additional resources
+//@subbarao, can you remind me why users should reference this doc. I recall you said something along the lines of continuous configuration or something
```
@subbarao-meduri or @bjoydeep

berenss replied:
This is an odd section - it's a bit of a grab-bag.
The first item `See [uuidegen](https://man7.org/linux/man-pages/man1/uuidgen.1.html) for instructions to create a UUID.` is relevant to `2.3.1. Designing your Grafana dashboard with a ConfigMap` or perhaps maybe it was related to an Observability troubleshooting item. 
The next item is a mention for the ensuing topic `2.6. Using managed cluster labels in Grafana`
And then the last 2 are navigational links back to the top of the sub-topic and back to the top of the Obs topic.  so I'm wondering if we just need to only keep the first bullet, about uuidegen, but move it to Troubleshooting.

dockerymick replied:
@berenss We are working on reducing the inline links, which can be distracting to the user. To work around that, I italicized "uuidegen" in the Designing your Grafana dashboard with a ConfigMap section. Using managed cluster labels in Grafana seemed like a related resource. The last two resources take the user back to the main topic that they were referenced from and to the beginning, in case the user chooses not to use the TOC 

Does that seem helpful? If not, I'm open to sharing and discussing. For the sake of the review of the TOC, I would like to make a comment about updating the Additional resources sections.

berenss replied:
Maybe move Additional Resources to the end of the sub topic?

dockerymick committed:
Apply suggestions from code review

dockerymick commented on clusters/cluster_lifecycle/single_node_ocp.adoc:
```diff
@@ -1,19 +1,19 @@
 = Creating a single node OpenShift cluster
 
-You can create a single node OpenShift (SNO) cluster that contains only a single node. In a standard cluster, there are control plane nodes and separate worker nodes. This configuration can be helpful when you have many clusters that need to minimize their resource footprint. 
+You can create a single-node OpenShift cluster that contains only a single node. In a standard cluster, there are control plane nodes and separate worker nodes. This configuration can be helpful when you have many clusters that need to minimize their resource footprint. 
```
add conref for single-node OpenShift. I suggest {sno}

dockerymick commented on clusters/cluster_lifecycle/single_node_ocp.adoc:
```diff
@@ -45,20 +39,21 @@ Add the `spec:skipMachinePools` parameter in the YAML content with a value of `t
 [#dynamic-metrics-for-sno]
-== Dynamic metrics for SNO clusters
+== Dynamic metrics for single-node OpenShift clusters
+//this information is also in the customize_observability.adoc file should this be removed from the Customize obseravbility file? Brandi: I agree that this is misplaced if it is only for RHACM uses, this i sitting in MCE doc.
```
@subbarao-meduri @berenss does the mce user have observability at this point or even knows what it is? If not, they might not use it. Do we need this information in both areas? Where does the user come across this first? Where will the user interact with this first

dockerymick replied:
Making a comment here again so that we don't lose the conversation

dockerymick replied:
@bjoydeep can you help confirm this since Subbarao is on PTO

berenss replied:
MCE does not include the RHACM Observability components. It is impossible for the MCE user to deploy Observability - they of course can deploy RHACM first, and then could deploy RHACM Observability. So we do not want any of this to be in an MCE area.

dockerymick replied:
Removed this section

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -277,7 +207,90 @@ spec:
+
+[#dynamic-metrics-for-sno]
+=== Dynamic metrics for single-node OpenShift clusters
+//recommend to remove this info and add reference link to the same section that is within the cluster lifecycle folder. | MJ | 09/29/23
```
@subbarao-meduri

berenss replied:
This must stay in the RHACM Observability section. It is only possible to do with RHACM and Observability.

dockerymick replied:
Updated

dockerymick committed:
Update use_observability.adoc

Updates during live review w/Brandi

dockerymick committed:
Update use_observability.adoc

dockerymick committed:
Update design_grafana.adoc

changes during live review

dockerymick committed:
Adding conref for single-node OpenShift

dockerymick committed:
Updating instances

dockerymick committed:
Update use_observability.adoc

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,204 @@
+
+* You can also access the `rbac-query-proxy` route with your OpenShift OAuth access token. The token should be associated with a user or service account, which has permission to get namespaces. For more information, see link:https://docs.openshift.com/container-platform/4.11/authentication/managing-oauth-access-tokens.html[Managing user-owned OAuth access tokens].
+//should this be a step
+//
```
```suggestion
//is this a substep if the user chooses to access the `rbac-query-proxy`?
```

dockerymick committed:
Update observability/use_observability.adoc

dockerymick committed:
Update design_grafana.adoc

updating stolostron link

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -97,32 +113,31 @@ metadata:
 If you have a managed cluster with monitoring for user-defined workloads enabled, the user workloads are located in the `test` namespace and generate metrics. These metrics are collected by Prometheus from the {ocp-short} user workload.
 
-Collect the metrics from the user workloads by creating a ConfigMap named, `observability-metrics-custom-allowlist` in the `test` namespace. View the following example:
+Add user workload metrics to the `observability-metrics-custom-allowlist` ConfigMape to collect the metrics in the `test` namespace. View the following example:
```
```suggestion
Add user workload metrics to the `observability-metrics-custom-allowlist` ConfigMap to collect the metrics in the `test` namespace. View the following example:
```

dockerymick committed:
Update observability/customize_observability.adoc

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -97,32 +113,31 @@ metadata:
 ----
-
++
+<1> In the `names` section, add the name of the custom metrics that is to be collected from the managed cluster.
```
```suggestion
<1>  Add the name of the custom metrics that are to be collected from the managed cluster.
```

dockerymick committed:
Update observability/customize_observability.adoc

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -97,32 +113,31 @@ metadata:
-
++
+<1>  Add the name of the custom metrics that are to be collected from the managed cluster.
+<2> In the `rules` section, enter only one value for the `expr` and `record` parameter pair to define the query expression. The metrics are collected as the name that is defined in the `record` parameter from your managed cluster. The metric value returned are the results after you run the query expression.
```
```suggestion
<2> *Optional:* Enter only one value for the `expr` and `record` parameter pair to define the query expression. The metrics are collected as the name that is defined in the `record` parameter from your managed cluster. The metric value returned are the results after you run the query expression.
```

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -97,32 +113,31 @@ metadata:
 ----
-
++
+<1>  Add the name of the custom metrics that are to be collected from the managed cluster.
```
```suggestion
<1> *Optional:* Add the name of the custom metrics that are to be collected from the managed cluster.
```

dockerymick commented on observability/customize_observability.adoc:
```diff
@@ -97,32 +113,31 @@ metadata:
+<1>  Add the name of the custom metrics that are to be collected from the managed cluster.
+<2> In the `rules` section, enter only one value for the `expr` and `record` parameter pair to define the query expression. The metrics are collected as the name that is defined in the `record` parameter from your managed cluster. The metric value returned are the results after you run the query expression.
++
+The `names` and `rules` sections are optional. You can use either one or both of the sections.
```
```suggestion
You can use either one or both of the sections.
```

dockerymick commented on observability/observe_environments_intro.adoc:
```diff
@@ -19,6 +18,10 @@ Read the following documentation for more details about the observability compon
 * xref:../observability/observability_arch.adoc#observability-arch[Observability architecture]
 * xref:../observability/observe_environments.adoc#observing-environments[Observability configuration]
+* xref:../observability/observability_enable#enabling-observability-service[Enabling the observability service]
+* xref:../observability/use_observability.adoc#using-observability[Using observability]
```
```suggestion
//looking to rename this section
* xref:../observability/use_observability.adoc#using-observability[Using observability]
```

dockerymick committed:
Update observability/observe_environments_intro.adoc

dockerymick committed:
Update use_observability.adoc

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,210 @@
+
+* You can also access the `rbac-query-proxy` route with your OpenShift OAuth access token. The token should be associated with a user or service account, which has permission to get namespaces. For more information, see link:https://docs.openshift.com/container-platform/4.11/authentication/managing-oauth-access-tokens.html[Managing user-owned OAuth access tokens].
+//should this be a step
+//is this a substep if the user chooses to access the `rbac-query-proxy`?
```
@bjoydeep i need your help confirming if these should be ordered steps or unordered options

berenss replied:
I think it is another optional way to get the route and token.

dockerymick replied:
Updated and added hidden comment

dockerymick committed:
Update observability/customize_observability.adoc

dockerymick committed:
Update single_node_ocp.adoc

Removed Dynamic metric for SNO clusters section and added a hidden comment for additional resources section

dockerymick committed:
Update customize_observability.adoc

Confirmed with PM that this should be removed:https://github.com/stolostron/rhacm-docs/pull/5520#discussion_r1349349128

dockerymick committed:
Update observability/customize_observability.adoc

dockerymick committed:
Update observability/customize_observability.adoc

dockerymick committed:
Update design_grafana.adoc

some updates and comments

dockerymick committed:
Update use_observability.adoc

updates

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,212 @@
+
+* You can also access the `rbac-query-proxy` route with your OpenShift OAuth access token. The token should be associated with a user or service account, which has permission to get namespaces. For more information, see link:https://docs.openshift.com/container-platform/4.11/authentication/managing-oauth-access-tokens.html[Managing user-owned OAuth access tokens].
+
+//need to improve the flow here; should these steps be in a separate section? How is this related to querying external queries?
```
@subbarao-meduri @bjoydeep @berenss

berenss replied:
This is about using Observability in a different way - rather than the user opening a browser and loading ACM UI and clicking the Grafana link to view metrics graphics, well in this case, the user has decided they want to use their trusty terminal window and grab metrics using CLI methods.
So this is about 'using obs' and it is just a different approach to get what they need.

dockerymick replied:
@berenss thanks for that explanation. i was referring to the lines 26-58. Should this be a subsection to Querying external metrics and named Creating secrets for observability proxy certificates ? Or Querying metrics from proxy certificates?

dockerymick committed:
Update use_observability.adoc

updating the sections

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,207 @@
+//after changes, let's reconsider the name for this topic. How will the user know what this is for? What is the content for? Implementing observability? Using observability is broad
+Use the observability service to view the utilization of clusters across your fleet.
+
+- <<external-metric-query,Using the external metric query>>
```
```suggestion
- <<external-metric-query,Querying external metrics>>
```

dockerymick committed:
Update observability/use_observability.adoc

dockerymick commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,207 @@
+*Note:* Do not set the `ObservabilitySpec.resources.CPU.limits` parameter if observability is enabled on single node managed clusters. When you set the CPU limits, it causes the observability pod to be counted against the capacity for your managed cluster. See the reference for _Management Workload Partitioning_ in the _Additional resources_ section.
+
+View the following dashboard options to include, RHACM dashboards, an ETCD table, or cluster service for Kubernetes API:
+[#acm-provided-dashboards]
```
```suggestion

[#acm-provided-dashboards]
```

dockerymick committed:
Update observability/use_observability.adoc

dockerymick commented on clusters/cluster_lifecycle/single_node_ocp.adoc:
```diff
@@ -22,19 +22,13 @@ This procedure explains how to deploy a single SNO cluster. You can also test th
+To create a {sno} cluster, complete the following steps in the {product-title-short} console for creating a cluster on one of the providers that are supported to run a {sno} cluster in xref:../cluster_lifecycle/create_intro.adoc#creating-a-cluster[Creating a cluster]:
 
-Complete the steps in the {product-title-short} console for creating a cluster on one of the providers that are supported to run a SNO cluster in xref:../cluster_lifecycle/create_intro.adoc#creating-a-cluster[Creating a cluster].
+. Select {ocp-short} 4.11, or later In the _Release image_ field. If you enabled the feature, an option to select *Single node OpenShift* is displayed. If you select an {ocp-short} version that is earlier than 4.11, the option to select _Single node OpenShift_ is not displayed. When you select *Single node OpenShift* during the creation process, the _Worker pools_ option is removed from the proce ...
```
```suggestion
. Select {ocp-short} 4.11, or later In the _Release image_ field. If you enabled the feature, an option to select *{sno}* is displayed. If you select an {ocp-short} version that is earlier than 4.11, the option to select _Single node OpenShift_ is not displayed. When you select *Single node OpenShift* during the creation process, the _Worker pools_ option is removed from the procedure because there are no worker nodes in a {sno} cluster. 
```

dockerymick committed:
Update clusters/cluster_lifecycle/single_node_ocp.adoc

dockerymick commented on clusters/cluster_lifecycle/single_node_ocp.adoc:
```diff
@@ -22,19 +22,13 @@ This procedure explains how to deploy a single SNO cluster. You can also test th
+To create a {sno} cluster, complete the following steps in the {product-title-short} console for creating a cluster on one of the providers that are supported to run a {sno} cluster in xref:../cluster_lifecycle/create_intro.adoc#creating-a-cluster[Creating a cluster]:
 
-Complete the steps in the {product-title-short} console for creating a cluster on one of the providers that are supported to run a SNO cluster in xref:../cluster_lifecycle/create_intro.adoc#creating-a-cluster[Creating a cluster].
+. Select {ocp-short} 4.11, or later In the _Release image_ field. If you enabled the feature, an option to select *{sno}* is displayed. If you select an {ocp-short} version that is earlier than 4.11, the option to select _Single node OpenShift_ is not displayed. When you select *Single node OpenShift* during the creation process, the _Worker pools_ option is removed from the procedure because the ...
```
```suggestion
. Select {ocp-short} 4.11, or later In the _Release image_ field. If you enabled the feature, an option to select *{sno}* is displayed. If you select an {ocp-short} version that is earlier than 4.11, the option to select _{sno}_ is not displayed. When you select *{sno}* during the creation process, the _Worker pools_ option is removed from the procedure because there are no worker nodes in a {sno} cluster. 
```

dockerymick committed:
Update clusters/cluster_lifecycle/single_node_ocp.adoc

dockerymick committed:
Update single_node_ocp.adoc

removing comment

dockerymick committed:
Update customize_observability.adoc

removed most hidden comments

dockerymick committed:
Update design_grafana.adoc

removed most of the hidden comments

dockerymick committed:
Update observability_enable.adoc

removing most of the hidden comments and replacing the OCP link

dockerymick committed:
Update observe_environments_intro.adoc

adding to the comment

swopebe commented on observability/observability_enable.adoc:
```diff
@@ -485,133 +476,68 @@ Create your object storage secret named, `thanos-object-storage` in the `open-cl
+- To learn more about customizing the observability service, see xref:../observability/customize_observability.adoc#customizing-observability[Customizing observability].
 
-To learn more about customizing the observability service, see xref:../observability/customize_observability.adoc#customizing-observability[Customizing observability].
+- For more related topics, return to the xref:../observability/observe_environments_intro.adoc#observing-environments-intro[Observability service introduction].
```
We aren't doing this last line consistently and not sure that we need to always have this at the end...

swopebe commented on observability/observe_environments_intro.adoc:
```diff
@@ -1,8 +1,7 @@
+The observability component is a service you can use to understand the health and utilization of clusters across your fleet. By default, multicluster observability operator is enabled during the installation of {product-title-short}. 
 
 [#observability-support]
 == Support
```
"Support"? We have a support section of the doc (troubleshooting) so I am trying to understand this intention of this section. Remember, we don't have to add everything to the doc just bc we are asked, we should consider the meaning...we are basically sending them to the doc with this topic. I would like us to move to simplifying the observability doc a bit more.

swopebe commented on observability/use_observability.adoc:
```diff
@@ -0,0 +1,208 @@
+
+. Add the `writeStorage` parameter to the `MultiClusterObservability` custom resource for adding a list of external endppoints that you want to export. View the following example:
++
+[source,yaml]
```
We have a format issue here, likely due to the table bold tags, which we really can remove and put in code. That is what some of the tables are doing in our doc. So `insecure_skip_verify`

dockerymick replied:
@swopebe thanks for the callout. I removed the extra asterisks

swopebe commented on release_notes/whats_new.adoc:
```diff
@@ -22,13 +23,8 @@
 
-//2.9:6549
-* Use the _Beta view_ switch from the _Overview_ page header for your Grafana metrics to be displayed, and filter the page data by using cluster labels. See, link:../console/console.adoc#console-overview[Console overview] for more details.
+//team said nothing new this release, if that is the case, we can remove any section with no updates
```
Let's remove this comment and make any notes in the what's new issue, to keep comments down. I will know to remove this if there is no update.

swopebe reviewed (approved):
Gonna approve even though I don't want to merge with these internal comments and I want to chat about the "support" section there. We can discuss in Monday's meeting if we have time.

dockerymick committed:
Update whats_new.adoc

removing hidden comments

dockerymick committed:
Merge branch '2.9_stage' into mj-ACM-6946

dockerymick committed:
Update use_observability.adoc

removing some hidden comments

dockerymick committed:
Update use_observability.adoc

Removed extra asterisks
